File: frontend/src/types/winnerDetection.ts
// Mirrors backend/app/schemas/winner_detection.py exactly -- see
// docs/features/73-winner-detection.md.

export interface WinnerGroupStats {
  dimension: string;
  label: string;

  sample_size: number;
  linked_sample_size: number;
  min_sample_size: number;
  meets_minimum_sample: boolean;
  confidence: "insufficient" | "low" | "medium" | "high";

  avg_views: number | null;
  median_views: number | null;
  avg_engagement_rate: number | null;
  avg_share_rate: number | null;
  avg_follower_conversion_rate: number | null;
  avg_views_per_day_since_publish: number | null;

  performance_score: number | null;
  performance_score_basis: string;
  note: string | null;
}

export interface TrendGroupStats {
  dimension: string;
  label: string;
  trend: "rising" | "underperforming" | "stable" | "insufficient_data";
  earlier_avg_score: number | null;
  recent_avg_score: number | null;
  change_pct: number | null;
  note: string;
}

File: backend/app/schemas/winner_detection.py
from pydantic import BaseModel, ConfigDict


class WinnerGroupStatsOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    dimension: str
    label: str

    sample_size: int
    linked_sample_size: int
    min_sample_size: int
    meets_minimum_sample: bool
    confidence: str

    avg_views: float | None
    median_views: float | None
    avg_engagement_rate: float | None
    avg_share_rate: float | None
    avg_follower_conversion_rate: float | None
    avg_views_per_day_since_publish: float | None

    performance_score: float | None
    performance_score_basis: str
    note: str | None


class TrendGroupStatsOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    dimension: str
    label: str
    trend: str
    earlier_avg_score: float | None
    recent_avg_score: float | None
    change_pct: float | None
    note: str
